hack/make-rules/run-e2e-tests.sh

```diff
@@ -25,20 +25,119 @@ source "${YURT_ROOT}/hack/lib/build.sh"
+    cd ${target_bin_dir}
+    echo "Building ${YURT_E2E_TARGETS}"
+    testpkg="$(dirname ${YURT_E2E_TARGETS})"
+    filename="$(basename ${YURT_E2E_TARGETS})"
+#    set kubeconfig
+    docker exec -d $edgeNodeContainerName /bin/bash -c 'mkdir /root/.kube/ -p'
+    docker exec -d $edgeNodeContainerName /bin/bash -c  "echo 'export KUBECONFIG=/root/.kube/config' >> /etc/profile && source /etc/profile" 
+    docker cp $KUBECONFIG $edgeNodeContainerName:/root/.kube/config
+}
+
+# set up flannel
+function setUpFlannel() {
+    local flannelYaml="https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml"
+    local flannelDs="kube-flannel-ds"
+    local flannelNameSpace="kube-flannel"
+    local POD_CREATE_TIMEOUT=120s
+    curl -o ${YURT_ROOT}/flannel.yaml $flannelYaml
+    kubectl apply -f ${YURT_ROOT}/flannel.yaml
+    # check if flannel on every node is ready, if so, "daemon set "kube-flannel-ds" successfully rolled out"
+    kubectl rollout status daemonset kube-flannel-ds -n kube-flannel --timeout=${POD_CREATE_TIMEOUT}
+
+    # set up bridge cni plugins for every node
+    wget -O ${YURT_ROOT}/cni.tgz https://github.com/containernetworking/plugins/releases/download/v1.1.1/cni-plugins-linux-amd64-v1.1.1.tgz
+
+    docker cp ${YURT_ROOT}/cni.tgz $cloudNodeContainerName:/opt/cni/bin/
+    docker exec -t $cloudNodeContainerName /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'
+
+    docker cp ${YURT_ROOT}/cni.tgz $edgeNodeContainerName:/opt/cni/bin/
+    docker exec -t $edgeNodeContainerName /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'
+
+    docker cp ${YURT_ROOT}/cni.tgz $edgeNodeContainer2Name:/opt/cni/bin/
+    docker exec -t $edgeNodeContainer2Name /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'    
+}
+
+# install gingko
+function getGinkgo() {
+    go install github.com/onsi/ginkgo/v2/ginkgo@v2.3.0
+    go get github.com/onsi/gomega/...
+    go mod tidy
+}
 
 # run e2e tests
-function run_e2e_tests {
+function run_non_edge_autonomy_e2e_tests {
     # check kubeconfig
-    if [ ! -f "${KUBECONFIG}" ]; then
-        echo "kubeconfig does not exist at ${KUBECONFIG}"
-        exit -1
-    fi
-
-    local target_bin_dir=$(get_binary_dir_with_arch ${YURT_LOCAL_BIN_DIR})
-    local e2e_test_file_name=$(basename ${YURT_E2E_TARGETS})
-    ${target_bin_dir}/${e2e_test_file_name} -kubeconfig ${KUBECONFIG}
+        if [ ! -f "${KUBECONFIG}" ]; then
+            echo "kubeconfig does not exist at ${KUBECONFIG}"
+            exit -1
+        fi
+    # run non-edge-autonomy-e2e-tests
+    cd $YURT_ROOT/test/e2e/
+    ginkgo --gcflags "${gcflags:-}" ${goflags} --label-filter='!edge-autonomy' -r
 }
 
-GOOS=${LOCAL_OS} GOARCH=${LOCAL_ARCH} build_e2e
-run_e2e_tests
\ No newline at end of file
+function run_e2e_edge_autonomy_tests {
+     # check kubeconfig
+        if [ ! -f "${KUBECONFIG}" ]; then
+            echo "kubeconfig does not exist at ${KUBECONFIG}"
+            exit -1
+        fi
```

Comment:
be cautious about indent style. there are too much prefix white space

Reply:
will fix